src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -404,9 +404,9 @@ bool Lowering::IsContainableUnaryOrBinaryOp(GenTree* parentNode, GenTree* childN
                 return false;
             }
 
-            if (childNode->gtGetOp1()->OperIs(GT_CAST))
+            if (childNode->gtGetOp1()->OperIs(GT_CAST) && !parentNode->gtGetOp2()->IsIntegralConst())
```

Comment:
```suggestion
            if (childNode->gtGetOp1()->OperIs(GT_CAST) && !parentNode->gtGetOp2()->isContained())
```
Should this check rather be like this? Uncontained constant will still be in a register, and that should be fine.

Reply:
Looking again, I think it was down to the cast node needing more restrictions. I thought the assertion `isGeneralRegister(reg2)` was firing on `REG_NA`, but it was actually firing on a floating point register. The other assertion is down to containing a cast node that already contains a `LCL_VAR`, this shouldn't be allowed.